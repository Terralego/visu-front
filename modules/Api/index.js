import Api from './services/api';

export * from './services/api';
export { default as log } from './services/log';

export { default as ApiProvider } from './components/ApiProvider';

export default Api;
