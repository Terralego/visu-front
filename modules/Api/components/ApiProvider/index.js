import Api from '../..';

export const ApiProvider = ({ host, children }) => {
  Api.host = host;
  return children;
};

export default ApiProvider;
