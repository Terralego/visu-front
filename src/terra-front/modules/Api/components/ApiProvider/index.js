import Api from '../../services/api';

export const ApiProvider = ({ host, children }) => {
  Api.host = host;
  return children;
};

export default ApiProvider;
