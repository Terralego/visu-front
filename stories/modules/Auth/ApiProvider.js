import Api from '../../../modules/Api';

export const ApiProvider = ({ host, children }) => {
  Api.host = host;
  return children;
};


export default ApiProvider;
