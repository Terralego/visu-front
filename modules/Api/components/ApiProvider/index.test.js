import React from 'react';
import renderer from 'react-test-renderer';

import Api from '../../';
import ApiProvider from './';

it('should render children correctly', () => {
  const Div = () => <div>Test</div>;
  const divRendered = renderer.create(<Div />).toJSON();
  const WrappedDiv = () => <ApiProvider host="host"><Div /></ApiProvider>;
  const wrappedDivRenderer = renderer.create(<WrappedDiv />).toJSON();

  expect(divRendered).toEqual(wrappedDivRenderer);
  expect(wrappedDivRenderer).toMatchSnapshot();
  expect(Api.host).toBe('host');
});
