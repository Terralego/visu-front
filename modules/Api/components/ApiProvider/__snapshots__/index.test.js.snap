// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render children correctly 1`] = `
<div>
  Test
</div>
`;
